### PR TITLE
docs: Explain exchange rate terminology.

### PR DESCRIPTION
**Pull request by daniel-wong-dfinity-org**
In particular, what does "X/Y" mean? What are "base" and "quote" assets?

In short in "X/Y", X is the "base" asset, and Y is the "quote" asset, and it tells you the price of X in terms of Y.

At least for me, this is really not self-explanatory.

### CODE REVIEW

File: src/ic-xrc-types/src/lib.rs [hunk 1]
```diff
@@ -19,13 +19,24 @@ pub struct Asset {
...
+/// completely backwards: you would say that the "rate" in this example is
+/// "30_000 USDT per BTC", which would be written as "30_000 USDT/BTC".
+/// Economists do NOT read "BTC/USD" as "BTC per USD". In this example, BTC is
+/// the "base" asset, and USDT is the "quote" asset.
+///
+/// Translation table for non-economists:
+///
+///             X/Y = Ys per X
+///      base asset = resource that you want to receive
+///     quote asset = resource that you are offerring
```
Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:34]: I like the intent but it's too verbose for my liking.
How about we just refer to the definition of [currency pair](https://en.wikipedia.org/wiki/Currency_pair)?
I think it's more important to point out that `X` is the base asset and `Y` is the quote asset in the expression `X/Y` and explain "base asset" and "quote asset". Of course, adding a small example may help.

File: src/ic-xrc-types/src/lib.rs [hunk 2]
```diff
@@ -19,13 +19,24 @@ pub struct Asset {
...
 #[derive(CandidType, Clone, Debug, Deserialize)]
 pub struct GetExchangeRateRequest {
-    /// The asset to be used as the resulting asset. For example, using
-    /// ICP/USD, ICP would be the base asset.
+    /// The resource being sought.
     pub base_asset: Asset,
-    /// The asset to be used as the starting asset. For example, using
-    /// ICP/USD, USD would be the quote asset.
+    /// The resource being offerred.
```
Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:39]: I don't think we should use words like "sought" and "offered".
I think if we define base asset and quote asset above, it should be clear here.

Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs]: Done.

# Tangent.

Today, I am under a new understanding (which might also be incorrect): X/Y does **not** indicate which direction you are going. You could just as well be going from X to Y, or from Y to X. The direction would be indicated elsewhere.

I also learned (thanks to the article that you shared) that X is the "more important" one, but there is no globally agreed upon ranking of the world's currencies.

File: src/ic-xrc-types/src/lib.rs [hunk 3]
```diff
@@ -19,13 +19,13 @@ pub struct Asset {
 }
 
 /// The type the user sends when requesting a rate.
+///
+/// For definitions of "base", "quoted", etc, refer to https://en.wikipedia.org/wiki/Currency_pair
```
Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:23]:
```suggestion
/// For definitions of "base", "quoted", etc, refer to https://en.wikipedia.org/wiki/Currency_pair.
```

File: src/ic-xrc-types/src/lib.rs [hunk 4]
```diff
@@ -19,13 +19,13 @@ pub struct Asset {
...
 #[derive(CandidType, Clone, Debug, Deserialize)]
 pub struct GetExchangeRateRequest {
-    /// The asset to be used as the resulting asset. For example, using
-    /// ICP/USD, ICP would be the base asset.
+    /// The base asset.
     pub base_asset: Asset,
-    /// The asset to be used as the starting asset. For example, using
-    /// ICP/USD, USD would be the quote asset.
+    /// The quoted asset.
```
Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:28]: While not explicitly mentioned in the currency-pair wiki page, it is usually called "quote asset" (and not "quoted asset"). Google Trends agrees with this.   :slightly_smiling_face:

File: src/ic-xrc-types/src/lib.rs [hunk 5]
```diff
@@ -56,7 +56,7 @@ pub struct ExchangeRateMetadata {
 pub struct ExchangeRate {
     /// The base asset.
     pub base_asset: Asset,
-    /// The quote asset.
+    /// The quoted asset.
```
Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:59]: Same here.

Comment (dskloetd) [reply src/ic-xrc-types/src/lib.rs:59]: Regardless of "quote" vs. "quoted", putting a comment `// The foo` on a field called `foo`, doesn't add any value IMO. Can we just remove this?

File: src/ic-xrc-types/src/lib.rs (hunk 4 above)
Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs]: Done.

File: src/ic-xrc-types/src/lib.rs (hunk 5 above)
Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs]: > doesn't add any value

👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍

FWIW, it was already like that.

I'd prefer to just delete it, but I don't own this, and deleting would be outside the original intent of this PR -> I do as Thomas says 🙇

File: src/ic-xrc-types/src/lib.rs (hunk 3 above)
Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs]: Done.

File: src/ic-xrc-types/src/lib.rs (hunk 5 above)
Comment (THLO) [reply src/ic-xrc-types/src/lib.rs]: :laughing: 
I agree with @dskloetd (and you). Feel free to delete these comment lines as part of this PR!

File: src/ic-xrc-types/src/lib.rs [hunk 6]
```diff
@@ -19,13 +19,14 @@ pub struct Asset {
...
 
 /// The type the user sends when requesting a rate.
+///
+/// For definitions of "base", "quote", etc, the reader is referred to
+/// https://en.wikipedia.org/wiki/Currency_pair.
 #[derive(CandidType, Clone, Debug, Deserialize)]
 pub struct GetExchangeRateRequest {
-    /// The asset to be used as the resulting asset. For example, using
-    /// ICP/USD, ICP would be the base asset.
+    /// The base asset.
```
Comment (dskloetd) [inline src/ic-xrc-types/src/lib.rs:27]: I think the comment that was here before had some value but now it's a comment that adds no value.

Actually, I think the example was more useful than a link to some long definition.

Comment (dskloetd) [reply src/ic-xrc-types/src/lib.rs:27]: As before, it's up to @THLO to agree or not.

Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:27]:
```suggestion
    /// The base asset, i.e., the first asset in a currency pair. For example, ICP is the base asset in the currency pair ICP/USD.
```

File: src/ic-xrc-types/src/lib.rs [hunk 7]
```diff
@@ -19,13 +19,14 @@ pub struct Asset {
...
     pub base_asset: Asset,
-    /// The asset to be used as the starting asset. For example, using
-    /// ICP/USD, USD would be the quote asset.
+    /// The quote asset.
```
Comment (THLO) [inline src/ic-xrc-types/src/lib.rs:29]:
```suggestion
    /// The quote asset, i.e., the second asset in a currency pair. For example, USD is the quote asset in the currency pair ICP/USD.
```

File: src/ic-xrc-types/src/lib.rs (hunk 6 above)
Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs:27]: Thomas made a suggestion. I'm going to take it.

Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs]: Done.

File: src/ic-xrc-types/src/lib.rs (hunk 7 above)
Comment (daniel-wong-dfinity-org) [reply src/ic-xrc-types/src/lib.rs]: Done.

**Review by THLO — APPROVED** [review]

**Pull request merged**